zanata-war/src/main/java/org/zanata/adapter/OkapiFilterAdapter.java

```diff
@@ -187,11 +191,17 @@ public Resource parseDocumentFile(URI documentContent,
                     if (!tu.getSource().isEmpty() && tu.isTranslatable()) {
                         String content = getTranslatableText(tu);
                         if (!content.isEmpty()) {
-                            TextFlow tf =
-                                    new TextFlow(getIdFor(tu, content,
-                                            subDocName), sourceLocale);
-                            tf.setPlural(false);
-                            tf.setContents(content);
+                            TextFlow tf = new TextFlow(getIdFor(tu, content,
+                                    subDocName), sourceLocale);
+                            // Numerus is the Qt TS plural indicator
+                            if(tu.hasProperty("numerus") &&
```

Comment:
I suggest extracting a method like makeTextFlow, calling it here, and overriding it in TSAdaptor with the custom logic.